Products/GenericSetup/tool.py

```diff
@@ -915,23 +915,52 @@ def profileExists(self, profile_id):
         else:
             return True
 
+    @security.protected(ManagePortal)
+    def profilesExist(self, profile_ids):
+        """Check if all profiles exist."""
+        for profile_id in profile_ids:
+            if not self.profileExists(profile_id):
+                return False
+        return True
+
+    security.declareProtected(ManagePortal, "dependenciesExist")
+    def dependenciesExist(self, profile_id):
+        """Check if all dependency-profiles exist."""
+        try:
+            dependency_ids = self.getDependenciesForProfile(profile_id)
+        except:
+            return False
+        return True
+
     @security.protected(ManagePortal)
     def getProfileInfo(self, profile_id):
         return _profile_registry.getProfileInfo(profile_id)
 
     @security.protected(ManagePortal)
     def getDependenciesForProfile(self, profile_id):
+
+        dependencies = None
+        profile_info = None
```

Comment:
Both variables are assigned unconditionally below -- no need to initialize them here.

Reply:
Oh yeah, thanks. Think I got overcautious, because it was the cause in the bug of CMFPlone.